src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -125,10 +150,15 @@ class jit_convert_array : public jit_kernel {
 
     template<typename src_t, typename dst_t>
     static fn_t get() {
-        if (mayiuse(cpu_isa_t::avx2)
-            && dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tF16C)) {
+        // fallback to AVX2 if AVX512 is not available
+        if (mayiuse(cpu_isa_t::avx512_core)) {
+            static jit_convert_array converter(convert_vec<src_t, dst_t>, sizeof(src_t), sizeof(dst_t));
+            auto& generator = static_cast<jit_generator&>(converter);
+            generator.create_kernel();
+            return (fn_t)generator.jit_ker();
+        } else if (mayiuse(cpu_isa_t::avx2) && dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tF16C)) {
             static jit_convert_array converter(convert_vec<src_t, dst_t>, sizeof(src_t), sizeof(dst_t));
-            auto & generator = static_cast<jit_generator&>(converter);
+            auto& generator = static_cast<jit_generator&>(converter);
             generator.create_kernel();
             return (fn_t)generator.jit_ker();
         }
```

Comment:
Looks the if and else branches are using the same code piece? How do you differentiate which specialized 
template is used?

Reply:
Probably we can use ISA as the template parameter here.